src/main/java/com/example/MyBatisDemo/services/StudentService.java

```diff
@@ -0,0 +1,49 @@
+package com.example.MyBatisDemo.services;
+
+import com.example.MyBatisDemo.dto.StudentAdditionDto;
+import com.example.MyBatisDemo.dto.StudentDisplayDto;
+import com.example.MyBatisDemo.dto.StudentSubjectsAdditionDto;
+import com.example.MyBatisDemo.dto.StudentSubjectsDisplayDto;
+import com.example.MyBatisDemo.mapstructMapper.StudentMapper;
+import com.example.MyBatisDemo.model.StudentEntity;
+import com.example.MyBatisDemo.repository.StudentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class StudentService {
+
+    @Autowired
+    StudentRepository studentRepository;
+
+    @Autowired
+    StudentMapper studentMapper;
+
+    public List<StudentDisplayDto> getAllStudents(String name, Integer age, String city){
+        List<StudentEntity> studentEntities = studentRepository.findAllStudents(name, age, city);
+        return studentEntities.stream().map(studentMapper::convertStudentEntityToStudentDisplayDto).collect(Collectors.toList());
```

Comment:
Instead of looping, we can mapper method that maps list of entities to list of dtos

Reply:
updated